src/rgw/rgw_rest_pubsub.cc

```diff
@@ -242,6 +246,18 @@ class RGWPSCreateTopicOp : public RGWOp {
 };
 
 void RGWPSCreateTopicOp::execute(optional_yield y) {
+  // master request will replicate the topic creation.
+  bufferlist indata;
+  if (!driver->is_meta_master()) {
+    op_ret = rgw_forward_request_to_master(
```

Comment:
shouldn't we create the persistent topic queue on all zones?

Reply:
same question for "remove"?

Reply:
> same question for "remove"?

Check the do-remove and do-add function inside the metadata handler. Svc-topic-rados.cc. We add/remove the persistent queue for secondary